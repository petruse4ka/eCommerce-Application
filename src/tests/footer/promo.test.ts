import { describe, expect, it } from 'vitest';

import Promo from '@/components/footer/promo';

describe('Snapshot test of promo section in footer', () => {
  it('should match the promo section snapshot', () => {
    const promo = new Promo();
    const element = promo.getElement();

    expect(element).toMatchSnapshot();
  });
});
