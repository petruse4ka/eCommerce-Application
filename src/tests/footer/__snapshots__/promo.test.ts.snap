// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer promo > should match the snapshot 1`] = `
<div
  class="flex flex-col items-center lg:items-start gap-3 lg:gap-6 w-full justify-between col-span-1"
>
  <h3
    class="text-xl font-bold text-center md:text-left"
  >
    Наши Преимущества
  </h3>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="Приготовлено c любовью"
      class="w-8 h-8"
      src="/src/assets/icons/footer-love.svg"
    />
    <span>
      Приготовлено c любовью
    </span>
  </div>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="Доставка в день заказа"
      class="w-8 h-8"
      src="/src/assets/icons/footer-delivery.svg"
    />
    <span>
      Доставка в день заказа
    </span>
  </div>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="100% миндальная мука"
      class="w-8 h-8"
      src="data:image/svg+xml,%3csvg%20width='29'%20height='37'%20viewBox='0%200%2029%2037'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M21.5656%207.07103C21.4872%206.94224%2021.3761%206.83561%2021.2432%206.76159C21.1102%206.68756%2020.96%206.64867%2020.8072%206.6487C20.6543%206.64874%2020.5041%206.6877%2020.3712%206.76179C20.2383%206.83588%2020.1272%206.94255%2020.0489%207.07138C18.7324%209.29108%2017.5379%2011.5784%2016.4706%2013.9235C14.72%209.2436%2012.5497%204.72548%209.98534%200.422302L9.98381%200.420414C9.94825%200.363924%209.90619%200.311639%209.85845%200.264589C9.84695%200.253027%209.8349%200.241464%209.82241%200.230711C9.77429%200.185694%209.72102%200.146287%209.66368%200.113274L9.66137%200.112542C9.60333%200.0811955%209.54184%200.056446%209.4781%200.0387729C9.462%200.0338781%209.4463%200.0297156%209.42981%200.025977C9.2969%20-0.00865901%209.15708%20-0.00865901%209.02417%200.025977C9.00768%200.0298312%208.99198%200.0336854%208.97584%200.0387729C8.91264%200.056267%208.85167%200.0807569%208.7941%200.111771L8.79018%200.113274C8.73283%200.146278%208.67956%200.185685%208.63145%200.230711C8.6188%200.241618%208.60691%200.252911%208.59541%200.264589C8.54767%200.311648%208.50559%200.363931%208.47001%200.420414L8.46848%200.422302C8.12298%200.989869%200%2014.4306%200%2023.2696C0%2033.3931%204.76649%2037%209.22699%2037C10.4029%2037%2011.5637%2036.7411%2012.6242%2036.2422C13.6846%2035.7434%2014.6175%2035.0174%2015.3541%2034.1177C15.9486%2035.004%2016.7582%2035.7315%2017.7101%2036.2346C18.6619%2036.7377%2019.7262%2037.0007%2020.8073%2037C24.5075%2037%2028.4615%2034.0303%2028.4615%2025.6949C28.4615%2018.4768%2021.8471%207.53322%2021.5656%207.07103ZM9.22699%2035.2641C8.80792%2035.1331%207.30001%2032.7608%207.30001%2023.2695C7.30001%2017.883%208.40688%2010.4675%209.22699%205.71228C10.0475%2010.4668%2011.1544%2017.8812%2011.1544%2023.2695C11.1544%2032.7623%209.64641%2035.1335%209.22699%2035.2641ZM1.76672%2023.2695C1.76672%2018.0302%204.96122%2010.7901%207.3675%206.07885C6.55198%2010.8812%205.53329%2017.944%205.53329%2023.2695C5.53329%2027.4424%205.81891%2032.1469%206.9116%2034.8243C3.68683%2033.5113%201.76672%2029.3323%201.76672%2023.2695ZM11.5423%2034.8243C12.6354%2032.1471%2012.9211%2027.4426%2012.9211%2023.2697C12.9211%2017.9442%2011.9024%2010.8817%2011.0869%206.07943C12.8592%209.47518%2014.3391%2013.0105%2015.5109%2016.6485L15.5113%2016.6504C15.8721%2017.8189%2016.1641%2019.0069%2016.3859%2020.2083C16.3893%2020.2279%2016.3928%2020.2482%2016.3962%2020.2678C16.4394%2020.5121%2016.4771%2020.7511%2016.5105%2020.9863C16.5151%2021.0195%2016.5204%2021.0526%2016.5251%2021.0857C16.5553%2021.3054%2016.5803%2021.5204%2016.6017%2021.7319C16.6059%2021.7733%2016.6117%2021.8151%2016.6155%2021.8562C16.6347%2022.0598%2016.6485%2022.2574%2016.66%2022.4531C16.6627%2022.4968%2016.6669%2022.5412%2016.6692%2022.5841C16.6807%2022.8182%2016.6872%2023.0471%2016.6872%2023.2695C16.6872%2023.4983%2016.6846%2023.7243%2016.6794%2023.9474C16.5969%2027.2972%2015.902%2030.0934%2014.6562%2032.0656C14.6461%2032.0849%2014.637%2032.1046%2014.6289%2032.1247C13.9149%2033.3209%2012.8348%2034.2655%2011.5423%2034.8241V34.8243ZM20.8073%2035.2656C18.98%2035.2656%2017.4774%2034.3356%2016.4369%2032.5711C17.5135%2030.6858%2018.2461%2028.1245%2018.4145%2024.7656C18.4184%2024.6892%2018.4214%2024.6128%2018.4248%2024.536C18.4421%2024.1243%2018.4539%2023.705%2018.4539%2023.2695C18.4539%2023.0139%2018.4461%2022.7512%2018.4321%2022.4825C18.4259%2022.3612%2018.4148%2022.2378%2018.406%2022.1151C18.3957%2021.9721%2018.3861%2021.8299%2018.3723%2021.683C18.3562%2021.5144%2018.3351%2021.3435%2018.3144%2021.1726C18.3021%2021.0729%2018.2908%2020.9739%2018.2772%2020.8723C18.2484%2020.6615%2018.2155%2020.4492%2018.1802%2020.2358C18.1702%2020.1749%2018.161%2020.1146%2018.1503%2020.0533C18.1062%2019.7973%2018.0578%2019.5396%2018.005%2019.2802L17.9935%2019.2237C17.8064%2018.3099%2017.5699%2017.3795%2017.2942%2016.4434C18.3286%2013.9912%2019.5015%2011.5976%2020.8073%209.27362C22.551%2012.3848%2026.6948%2020.3428%2026.6948%2025.6949C26.6948%2031.5985%2024.4389%2035.2656%2020.8073%2035.2656Z'%20fill='%234E9DD3'/%3e%3c/svg%3e"
    />
    <span>
      100% миндальная мука
    </span>
  </div>
</div>
`;

exports[`Snapshot test of promo section in footer > should match the promo section snapshot 1`] = `
<div
  class="flex flex-col items-center lg:items-start gap-3 lg:gap-6 w-full justify-between col-span-1"
>
  <h3
    class="text-xl font-bold text-center md:text-left"
  >
    Наши Преимущества
  </h3>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="Приготовлено c любовью"
      class="w-8 h-8"
      src="/src/assets/icons/footer-love.svg"
    />
    <span>
      Приготовлено c любовью
    </span>
  </div>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="Доставка в день заказа"
      class="w-8 h-8"
      src="/src/assets/icons/footer-delivery.svg"
    />
    <span>
      Доставка в день заказа
    </span>
  </div>
  <div
    class="flex flex-row text-[14px] items-center justify-center lg:justify-start text-center gap-2 w-full"
  >
    <img
      alt="100% миндальная мука"
      class="w-8 h-8"
      src="data:image/svg+xml,%3csvg%20width='29'%20height='37'%20viewBox='0%200%2029%2037'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M21.5656%207.07103C21.4872%206.94224%2021.3761%206.83561%2021.2432%206.76159C21.1102%206.68756%2020.96%206.64867%2020.8072%206.6487C20.6543%206.64874%2020.5041%206.6877%2020.3712%206.76179C20.2383%206.83588%2020.1272%206.94255%2020.0489%207.07138C18.7324%209.29108%2017.5379%2011.5784%2016.4706%2013.9235C14.72%209.2436%2012.5497%204.72548%209.98534%200.422302L9.98381%200.420414C9.94825%200.363924%209.90619%200.311639%209.85845%200.264589C9.84695%200.253027%209.8349%200.241464%209.82241%200.230711C9.77429%200.185694%209.72102%200.146287%209.66368%200.113274L9.66137%200.112542C9.60333%200.0811955%209.54184%200.056446%209.4781%200.0387729C9.462%200.0338781%209.4463%200.0297156%209.42981%200.025977C9.2969%20-0.00865901%209.15708%20-0.00865901%209.02417%200.025977C9.00768%200.0298312%208.99198%200.0336854%208.97584%200.0387729C8.91264%200.056267%208.85167%200.0807569%208.7941%200.111771L8.79018%200.113274C8.73283%200.146278%208.67956%200.185685%208.63145%200.230711C8.6188%200.241618%208.60691%200.252911%208.59541%200.264589C8.54767%200.311648%208.50559%200.363931%208.47001%200.420414L8.46848%200.422302C8.12298%200.989869%200%2014.4306%200%2023.2696C0%2033.3931%204.76649%2037%209.22699%2037C10.4029%2037%2011.5637%2036.7411%2012.6242%2036.2422C13.6846%2035.7434%2014.6175%2035.0174%2015.3541%2034.1177C15.9486%2035.004%2016.7582%2035.7315%2017.7101%2036.2346C18.6619%2036.7377%2019.7262%2037.0007%2020.8073%2037C24.5075%2037%2028.4615%2034.0303%2028.4615%2025.6949C28.4615%2018.4768%2021.8471%207.53322%2021.5656%207.07103ZM9.22699%2035.2641C8.80792%2035.1331%207.30001%2032.7608%207.30001%2023.2695C7.30001%2017.883%208.40688%2010.4675%209.22699%205.71228C10.0475%2010.4668%2011.1544%2017.8812%2011.1544%2023.2695C11.1544%2032.7623%209.64641%2035.1335%209.22699%2035.2641ZM1.76672%2023.2695C1.76672%2018.0302%204.96122%2010.7901%207.3675%206.07885C6.55198%2010.8812%205.53329%2017.944%205.53329%2023.2695C5.53329%2027.4424%205.81891%2032.1469%206.9116%2034.8243C3.68683%2033.5113%201.76672%2029.3323%201.76672%2023.2695ZM11.5423%2034.8243C12.6354%2032.1471%2012.9211%2027.4426%2012.9211%2023.2697C12.9211%2017.9442%2011.9024%2010.8817%2011.0869%206.07943C12.8592%209.47518%2014.3391%2013.0105%2015.5109%2016.6485L15.5113%2016.6504C15.8721%2017.8189%2016.1641%2019.0069%2016.3859%2020.2083C16.3893%2020.2279%2016.3928%2020.2482%2016.3962%2020.2678C16.4394%2020.5121%2016.4771%2020.7511%2016.5105%2020.9863C16.5151%2021.0195%2016.5204%2021.0526%2016.5251%2021.0857C16.5553%2021.3054%2016.5803%2021.5204%2016.6017%2021.7319C16.6059%2021.7733%2016.6117%2021.8151%2016.6155%2021.8562C16.6347%2022.0598%2016.6485%2022.2574%2016.66%2022.4531C16.6627%2022.4968%2016.6669%2022.5412%2016.6692%2022.5841C16.6807%2022.8182%2016.6872%2023.0471%2016.6872%2023.2695C16.6872%2023.4983%2016.6846%2023.7243%2016.6794%2023.9474C16.5969%2027.2972%2015.902%2030.0934%2014.6562%2032.0656C14.6461%2032.0849%2014.637%2032.1046%2014.6289%2032.1247C13.9149%2033.3209%2012.8348%2034.2655%2011.5423%2034.8241V34.8243ZM20.8073%2035.2656C18.98%2035.2656%2017.4774%2034.3356%2016.4369%2032.5711C17.5135%2030.6858%2018.2461%2028.1245%2018.4145%2024.7656C18.4184%2024.6892%2018.4214%2024.6128%2018.4248%2024.536C18.4421%2024.1243%2018.4539%2023.705%2018.4539%2023.2695C18.4539%2023.0139%2018.4461%2022.7512%2018.4321%2022.4825C18.4259%2022.3612%2018.4148%2022.2378%2018.406%2022.1151C18.3957%2021.9721%2018.3861%2021.8299%2018.3723%2021.683C18.3562%2021.5144%2018.3351%2021.3435%2018.3144%2021.1726C18.3021%2021.0729%2018.2908%2020.9739%2018.2772%2020.8723C18.2484%2020.6615%2018.2155%2020.4492%2018.1802%2020.2358C18.1702%2020.1749%2018.161%2020.1146%2018.1503%2020.0533C18.1062%2019.7973%2018.0578%2019.5396%2018.005%2019.2802L17.9935%2019.2237C17.8064%2018.3099%2017.5699%2017.3795%2017.2942%2016.4434C18.3286%2013.9912%2019.5015%2011.5976%2020.8073%209.27362C22.551%2012.3848%2026.6948%2020.3428%2026.6948%2025.6949C26.6948%2031.5985%2024.4389%2035.2656%2020.8073%2035.2656Z'%20fill='%234E9DD3'/%3e%3c/svg%3e"
    />
    <span>
      100% миндальная мука
    </span>
  </div>
</div>
`;
